File: backend/usport/src/main/java/com/app/usport/sport/SportService.java
package com.app.usport.sport;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
@Service
public class SportService {
    private SportRepository sportRepository;

    @Autowired
    public SportService(SportRepository sportRepository) {
        this.sportRepository = sportRepository;
    }

    public List<Sport> getAllSports() {
        return sportRepository.getAllSports();
    }

    public Sport getSport(String id) {
        return sportRepository.getSport(id);
    }

    public List<Sport> getSportsForField(String fieldId){
        return sportRepository.getSportsForField(fieldId);
    }
    public List<Sport> getSportsForUserMetric(String userMetricId){
        return sportRepository.getSportsForUserMetric(userMetricId);
    }
}

File: frontend/usport/src/services/SportService.js
import axios from 'axios';

class SporService{

    getAllSports(){
        return axios.get('http://localhost:8080/api/sport');
    }

    getSportsForField(fieldId){
        return axios.get('http://localhost:8080/api/sport?fieldId='+fieldId);
    }

    getSportsForUserMetric(userMetricId){
        return axios.get('http://localhost:8080/api/sport?userMetricId='+userMetricId);
    }
}

const sporService = new SporService();

export default sporService;
